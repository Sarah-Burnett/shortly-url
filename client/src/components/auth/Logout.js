import React from "react";
import { Link } from "react-router-dom";
import { AuthForm, PrimaryFormBtn } from "../Styles";
import { useUser } from "../../hooks/useUser";
import { useFlash } from "../../hooks/useFlash";
import { setAuthHeader } from "../../utilities/setAuthHeader";
import { removeLocalStorage } from "../../utilities/setLocalStorage";
import { useLinks } from "../../hooks/useLinks";

const Logout = () => {
	const { logoutUser } = useUser();
	const { showFlash } = useFlash();
	const { removeBrowserLinks } = useLinks();
	const logout = () => {
		setAuthHeader();
		removeLocalStorage("user");
		logoutUser();
		showFlash("Logout successful");
		removeBrowserLinks();
	};
	return (
		<AuthForm data-testid="logout">
			<h3>Are you sure you want to logout?</h3>
			<PrimaryFormBtn onClick={logout}>Yes</PrimaryFormBtn>
			<PrimaryFormBtn>
				<Link to="/dashboard">No</Link>
			</PrimaryFormBtn>
		</AuthForm>
	);
};

export default Logout;
