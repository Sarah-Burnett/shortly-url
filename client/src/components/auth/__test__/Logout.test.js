import {
	rendersWithoutCrashing,
	matchesSnapshot,
	ComponentWrapper,
} from "../../../setupTests";
import Logout from "../Logout";

const LogoutWrapper = () => ComponentWrapper(Logout);

it("renders logout without crashing", () =>
	rendersWithoutCrashing(LogoutWrapper));

it("logout matches snapshot", () => matchesSnapshot(LogoutWrapper));
