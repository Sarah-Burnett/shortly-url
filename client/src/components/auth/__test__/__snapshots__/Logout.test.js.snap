// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logout matches snapshot 1`] = `
<form
  className="sc-AxhUy dLXkxw"
  data-testid="logout"
>
  <h3>
    Are you sure you want to logout?
  </h3>
  <button
    className="sc-AxirZ sc-AxheI cOblFm"
    onClick={[Function]}
  >
    Yes
  </button>
  <button
    className="sc-AxirZ sc-AxheI cOblFm"
  >
    <a
      href="/dashboard"
      onClick={[Function]}
    >
      No
    </a>
  </button>
</form>
`;
